watertap/unit_models/ion_exchange_0D.py

```diff
@@ -0,0 +1,1352 @@
+        )
+
+        self.t_contact = Var(
+            initialize=520,
+            bounds=(120, None),
+            units=pyunits.s,
+            doc="Contact time [s]",
+        )
+
+        self.t_waste = Var(
+            initialize=12,
+            bounds=(1, None),
+            units=pyunits.s,
+            doc="Regen + Rinse + Backwash time [s]",
+        )
+
+        self.num_transfer_units = Var(
+            initialize=1e6,
+            bounds=(0, None),
+            units=pyunits.dimensionless,
+            doc="Number of transfer units [-]",
+        )
+
+        self.HTU = Var(
+            ion_set,
+            initialize=0.5,
+            bounds=(0, 1),
+            units=pyunits.m,
+            doc="Height of a transfer unit",
+        )
+
+        self.dimensionless_time = Var(
+            initialize=1,
+            bounds=(0, 1),
+            units=pyunits.dimensionless,
+            doc="Dimensionless time [-]",
+        )
+
+        self.lh = Var(
+            initialize=0,
+            bounds=(-20, 20),
+            units=pyunits.dimensionless,
+            doc="Left hand side of kinetic equation",
+        )
+
+        self.mass_in = Var(
+            ion_set,
+            initialize=1e3,
+            bounds=(0, None),
+            units=pyunits.mol,
+            doc="Influent mass of ion",
+        )
+
+        self.mass_removed = Var(
+            ion_set,
+            initialize=1e3,
+            bounds=(0, None),
+            units=pyunits.mol,
+            doc="Sorbed mass of ion",
+        )
+
+        self.mass_out = Var(
+            ion_set,
+            initialize=1e3,
+            bounds=(0, None),
+            units=pyunits.mol,
+            doc="Effluent mass of ion",
+        )
+
+        # # ====== Hydrodynamic variables ====== #
+
+        self.vel_min = Var(
+            initialize=0.0048,
+            bounds=(0, 0.009),
+            units=pyunits.m / pyunits.s,
+            doc="Minimum bed velocity [m/s]",
+        )
+
+        self.vel_bed = Var(
+            initialize=0.0086,
+            bounds=(0, 0.01),  # MWH, Perry's
+            units=pyunits.m / pyunits.s,
+            doc="Velocity through resin bed [m/s]",
+        )
+
+        self.vel_inter = Var(
+            initialize=0.0173,
+            bounds=(0, 0.07),
+            units=pyunits.m / pyunits.s,
+            doc="Interstitial velocity [m/s]",
+        )
+
+        self.holdup = Var(
+            initialize=100,
+            bounds=(90, 250),  # IX1
+            units=pyunits.dimensionless,
+            doc="Holdup percent [-]",
+        )
+
+        self.sfr = Var(
+            initialize=5,
+            bounds=(4, 40),
+            units=pyunits.hr**-1,
+            doc="Service flow rate [BV/hr]",
+        )
+
+        self.pressure_drop = Var(
+            initialize=14,
+            units=pyunits.psi,
+            bounds=(0, 25),
+            doc="Pressure drop across column",  # max pressure drop is 25 psi, MWH
+        )
```

Comment:
If you use a 0D control volume block from IDAES, you probably don't need separate variables for hold_up and pressure_drop.